bofire/domain/domain.py

```diff
@@ -266,9 +292,9 @@ def get_features(
         Returns:
             List[Feature]: List of features in the domain fitting to the passed requirements.
         """
-        return (self.input_features + self.output_features).get(
-            includes, excludes, exact
-        )
+        assert isinstance(self.input_features, InputFeatures)
```

Comment:
Why do we need this assert?

Reply:
Because the type annotation of `self.input_features` is `Union[Sequence[InputFeature], InputFeatures]` and not `InputFeatures`. If it was only `InputFeatures` Pyright would complain when you, e.g., construct via `Domain(input_features=[f1, f2], ...)`. The reason is that the static type checkers cannot keep track of the conversion of the Pydantic validators for `Sequence[InputFeature]` to `InputFeatures`.

Reply:
ah and the type checker gets it via the assert?

Reply:
Exactly.